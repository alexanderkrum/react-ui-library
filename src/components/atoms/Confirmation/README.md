# Configurable React Success Popup

## Screenshot with default settings

![alt screenshot](https://github.com/alexanderkrum/react-ui-library/blob/master/src/components/atoms/Confirmation/success-popup-small.png)

## Usage

```jsx
import SuccessPopup from 'react-success-popup'
;<SuccessPopup>Subscribed</SuccessPopup>
```

## Configuration

> All the CSS settings can be overridden. SVG animation accept only limited
> settings.

### Card settings

> its a div element so you can apply any settings you want.

The defaults:

```js
{
  transition: `opacity '300ms' ease-in`,
  opacity: opacity,
  position: 'absolute',
  left: '50vw',
  top: '50vh',
  transform: 'translate(-50%, -50%)',
  background: '#242424',
  borderRadius: '9px',
  width: '495px',
  height: '495px',
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
}
```

Example apply red background color to the popup

```jsx
<SuccessPopUp theme={card:{
    backgroundColor: 'red'
}}>Subscribed</SuccessPopUp>
```

### Check SVG settings

> its an animated svg element. Only three settings will be applied to that
> element.

The defaults:

```js
{
  strokeColor: '#B0B0B0',
  strokeWidth: 20,
  duration: '0.5s'
}
```

Example apply 1s check animation

```jsx
<SuccessPopUp theme={check:{
    duration: '1s'
}}>Subscribed</SuccessPopUp>
```

### Text wrapper settings

> Its a div wrapper around the span text. You can override and extend any
> setting.

The defaults:

```js
textWrapper: {
   marginTop: '75px',
   marginLeft: '-20px',
}
```

Example move text closer to the check

```jsx
<SuccessPopUp theme={textWrapper:{
    marginTop: '25px'
}}>Subscribed</SuccessPopUp>
```

### Text settings

> Its a span text. You can override and extend any setting.

The defaults:

```js
text: {
   fontFamily: 'Roboto',
   fontStyle: 'normal',
   fontWeight: '500',
   fontSize: '45px',
   lineHeight: '53px',
   color: '#AEAEAE',
}
```

Example make text of a red color

```jsx
<SuccessPopUp theme={text:{
    color: 'red'
}}>Subscribed</SuccessPopUp>
```
