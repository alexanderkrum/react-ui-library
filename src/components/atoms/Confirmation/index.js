// eslint-disable-next-line import/no-extraneous-dependencies
import React, { useEffect, useState } from 'react'

const Check = ({ theme }) => {
  return (
    <svg
      width="199"
      height="133"
      viewBox="0 0 199 133"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="10"
        y1="67"
        x2="15"
        y2="70.96"
        opacity="0"
        stroke={(theme && theme.strokeColor) || '#B0B0B0'}
        strokeWidth={(theme && theme.strokeWidth) || '20'}
        strokeLinecap="round"
      >
        <animate
          attributeName="x2"
          values="10;15;20;25;30;35;40;45;50;55;60;65;70;73"
          dur={(theme && theme.duration) || '0.5s'}
          repeatCount="1"
          fill="freeze"
        />
        <animate
          attributeName="y2"
          values="67;70.96;74.92;78.88;82.84;86.80;90.76;94.72;98.68;102.64;106.608;110.56;114.53;118.48;122.45"
          dur={(theme && theme.duration) || '0.5s'}
          repeatCount="1"
          fill="freeze"
        />
        <animate
          attributeName="opacity"
          from="0"
          to="1"
          dur="0.1s"
          fill="freeze"
        />
      </line>
      <line
        x1="73"
        y1="123"
        x2="73"
        y2="123"
        opacity="0"
        stroke={(theme && theme.strokeColor) || '#B0B0B0'}
        strokeWidth="20"
        strokeLinecap="round"
      >
        <animate
          attributeName="x2"
          values="73;83;93;103;113;123;133;143;153;163;173;183;188.008"
          begin={(theme && theme.duration) || '0.5s'}
          repeatCount="1"
          fill="freeze"
          dur={(theme && theme.duration) || '0.5s'}
        />
        <animate
          attributeName="y2"
          values="123;113;103;93;83;73;63;53;43;33;23;13;10"
          begin={(theme && theme.duration) || '0.5s'}
          repeatCount="1"
          fill="freeze"
          dur={(theme && theme.duration) || '0.5s'}
        />
        <animate
          attributeName="opacity"
          values="1"
          begin={(theme && theme.duration) || '0.5s'}
          dur="0s"
          fill="freeze"
        />
      </line>
    </svg>
  )
}

const Confirmation = ({ theme, children }) => {
  const [opacity, setOpacity] = useState(0)
  useEffect(() => {
    setOpacity(1)
  }, [])

  return (
    <div
      style={{
        ...{
          transition: `opacity '300ms' ease-in`,
          opacity,
          position: 'absolute',
          left: '50vw',
          top: '50vh',
          transform: 'translate(-50%, -50%)',
          background: '#242424',
          borderRadius: '9px',
          width: '495px',
          height: '495px',
          display: 'flex',
          flexDirection: 'column',
          justifyContent: 'center',
          alignItems: 'center',
        },
        ...(theme && theme.card),
      }}
    >
      <div>
        <Check theme={theme && theme.check} />
      </div>
      <div
        style={{
          ...{
            marginTop: '75px',
            marginLeft: '-20px',
          },
          ...(theme && theme.textWrapper),
        }}
      >
        <span
          style={{
            ...{
              fontFamily: 'Roboto',
              fontStyle: 'normal',
              fontWeight: '500',
              fontSize: '45px',
              lineHeight: '53px',
              color: '#AEAEAE',
            },
            ...(theme && theme.text),
          }}
        >
          {children}
        </span>
      </div>
    </div>
  )
}

export default Confirmation
