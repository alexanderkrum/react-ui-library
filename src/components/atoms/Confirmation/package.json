{
  "name": "react-success-popup",
  "version": "1.0.5",
  "description": "A configurable modal for success message",
  "main": "index.js",
  "scripts": {
    "test": "jest index.test.js"
  },
  "author": "alexanderkrum",
  "license": "ISC",
  "keywords": [
    "success",
    "popup",
    "popup configurable",
    "react",
    "animated svg"
  ]
}
