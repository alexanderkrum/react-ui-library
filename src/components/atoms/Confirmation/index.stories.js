// eslint-disable-next-line import/no-extraneous-dependencies
import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import { storiesOf } from '@storybook/react'
import Confirmation from './index'

storiesOf('Atoms/Confirmation', module)
  .add('default', () => <Confirmation>Subscribed</Confirmation>)
  .add('1s ease in duration', () => (
    <Confirmation
      theme={{ card: { easeInDuration: '1s' }, text: {}, check: {} }}
    >
      Subscribed
    </Confirmation>
  ))
  .add('white background', () => (
    <Confirmation
      theme={{ card: { backgroundColor: 'white' }, text: {}, check: {} }}
    >
      Subscribed
    </Confirmation>
  ))
  .add('red text and stroke', () => (
    <Confirmation
      theme={{
        card: { backgroundColor: 'white' },
        text: { color: 'red' },
        check: { strokeColor: 'red' },
      }}
    >
      Subscribed
    </Confirmation>
  ))
  .add('3 second animation', () => (
    <Confirmation theme={{ card: {}, text: {}, check: { duration: '3s' } }}>
      Subscribed
    </Confirmation>
  ))
