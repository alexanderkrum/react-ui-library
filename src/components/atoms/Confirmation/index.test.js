// eslint-disable-next-line import/no-extraneous-dependencies
import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react'
import Confirmation from './index'

test('it matches the snapshot', () => {
  const { container } = render(<Confirmation />)
  expect(container).toMatchSnapshot()
})
