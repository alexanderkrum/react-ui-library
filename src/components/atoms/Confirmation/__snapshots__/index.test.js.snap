// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
<div>
  <div
    style="transition: opacity '300ms' ease-in; opacity: 1; position: absolute; left: 50vw; top: 50vh; transform: translate(-50%, -50%); background: rgb(36, 36, 36); border-radius: 9px; width: 495px; height: 495px; display: flex; flex-direction: column; justify-content: center; align-items: center;"
  >
    <div>
      <svg
        fill="none"
        height="133"
        viewBox="0 0 199 133"
        width="199"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          opacity="0"
          stroke="#B0B0B0"
          stroke-linecap="round"
          stroke-width="20"
          x1="10"
          x2="15"
          y1="67"
          y2="70.96"
        >
          <animate
            attributeName="x2"
            dur="0.5s"
            fill="freeze"
            repeatCount="1"
            values="10;15;20;25;30;35;40;45;50;55;60;65;70;73"
          />
          <animate
            attributeName="y2"
            dur="0.5s"
            fill="freeze"
            repeatCount="1"
            values="67;70.96;74.92;78.88;82.84;86.80;90.76;94.72;98.68;102.64;106.608;110.56;114.53;118.48;122.45"
          />
          <animate
            attributeName="opacity"
            dur="0.1s"
            fill="freeze"
            from="0"
            to="1"
          />
        </line>
        <line
          opacity="0"
          stroke="#B0B0B0"
          stroke-linecap="round"
          stroke-width="20"
          x1="73"
          x2="73"
          y1="123"
          y2="123"
        >
          <animate
            attributeName="x2"
            begin="0.5s"
            dur="0.5s"
            fill="freeze"
            repeatCount="1"
            values="73;83;93;103;113;123;133;143;153;163;173;183;188.008"
          />
          <animate
            attributeName="y2"
            begin="0.5s"
            dur="0.5s"
            fill="freeze"
            repeatCount="1"
            values="123;113;103;93;83;73;63;53;43;33;23;13;10"
          />
          <animate
            attributeName="opacity"
            begin="0.5s"
            dur="0s"
            fill="freeze"
            values="1"
          />
        </line>
      </svg>
    </div>
    <div
      style="margin-top: 75px; margin-left: -20px;"
    >
      <span
        style="font-family: Roboto; font-style: normal; font-weight: 500; font-size: 45px; line-height: 53px; color: rgb(174, 174, 174);"
      />
    </div>
  </div>
</div>
`;
